source/docs/advanced/features/debugging.md

```diff
@@ -0,0 +1,98 @@
+
+Any debugger that can attach to Node.js can also be used to debug a Front-Commerce application. You can find more details in the Node.js [Debugging Guide](https://nodejs.org/en/docs/guides/debugging-getting-started/).
+
+## Debugging with VS Code
+
+Create a file named `.vscode/launch.json` at the root of your project with the following content:
+
+```json
+{
+  "version": "0.2.0",
+  "configurations": [
+    {
+      "name": "FC: debug server-side",
+      "type": "node-terminal",
+      "request": "launch",
+      "command": "npm run start -- --inspect"
+    },
+    {
+      "name": "FC: debug client-side",
+      "type": "pwa-chrome",
+      "request": "launch",
+      "url": "http://localhost:4000"
+    },
+    {
+      "name": "FC: debug full stack",
+      "type": "node-terminal",
+      "request": "launch",
+      "command": "npm run start -- --inspect",
+      "console": "integratedTerminal",
+      "serverReadyAction": {
+        "pattern": "started server on .+, url: (https?://.+)",
+        "uriFormat": "%s",
+        "action": "debugWithChrome"
+      }
+    }
+  ]
+}
+```
+
+If you're [changing the port number](/docs/reference/environment-variables.html#Host) your application starts on, replace the `3000` in `http://localhost:3000` with the port you're using instead, you can use the the url defined in your `FRONT_COMMERCE_URL`
+
+Now in vscode go to the Debug panel (<kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>D</kbd> on Windows/Linux, <kbd>⇧</kbd>+<kbd>⌘</kbd>+<kbd>D</kbd> on macOS), select a launch configuration, then press <kbd>F5</kbd> or select **Debug: Start Debugging** from the Command Palette to start your debugging session.
+
+## Debugging with Chrome DevTools
+
+### Client-side code
+
+Start your development server as usual by running `front-commerce start`, `npm run start`. Once the server starts, open `http://localhost:4000` (or your alternate URL) in Chrome. Next, open Chrome's Developer Tools (<kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>J</kbd> on Windows/Linux, <kbd>⌥</kbd>+<kbd>⌘</kbd>+<kbd>I</kbd> on macOS), then go to the **Sources** tab.
+
+Now, any time your client-side code reaches a [`debugger`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/debugger) statement, code execution will pause and that file will appear in the debug area. You can also press <kbd>Ctrl</kbd>+<kbd>P</kbd> on Windows/Linux or <kbd>⌘</kbd>+<kbd>P</kbd> on macOS to search for a file and set breakpoints manually.
+
+### Server-side code
+
+To debug server-side Front-Commerce code with Chrome DevTools, you need to pass the [`--inspect`](https://nodejs.org/api/cli.html#cli_inspect_host_port) flag to the underlying Node.js process:
```

Comment:
The docs are specifically aimed at either VS code debugger or chrome. dev tools (nodejs debugger), I will add a video example to show the diff between just a browser-based debugger (chrome, firefox, etc..) and nodejs debugger inside chrome.

Reply:
I added an example for both VSCode & Chrome DevTools

Reply:
I know Firefox does not support that but as mentioned in https://nodejs.org/en/docs/guides/debugging-getting-started/#inspector-clients there are others debuggers than Chrome dev tools (VSCode is one of them btw).

Reply:
I think you misunderstand my point. My suggestion above was just a little rephrasing so that the sentence is a bit more open. The way it is written seems to suggest that only Chrome Dev Tools can be used which is wrong. Any debugger implementing the correct protocol can be used.

Reply:
> also some months ago, I added a paragaph about debugging with the debugger in https://developers.front-commerce.com/docs/reference/environment-variables.html#Debugging so it would be nice to unify that with what you add. But besides this and the remarks 👍

Hmm, should I maybe move this debugging to that documentation?

Reply:
Added back links between the two docs